### PR TITLE
fixes deletion by preventing accessing stack output values

### PR DESCRIPTION
**Pull request by xh3b4sd**
Fixes https://github.com/giantswarm/giantswarm/issues/4651.

### CODE REVIEW

File: service/controller/v18/resource/cloudformation/current.go [hunk 1]
```diff
@@ -56,8 +56,22 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
...
 		}
+
+		// When a tenant cluster is deleted it might be not completely created yet
+		// in the first place. There can be issues with unaccessible stack output
+		// values in such cases, causing the deletion process to get into a
+		// deadlock. To remedy such cases we simply return the stack state
+		// containing the stack name, without trying to access any stack output
+		// values.
+		currentState := StackState{
+			Name: key.MainGuestStackName(customObject),
```
Comment (MarcelMue) [inline service/controller/v18/resource/cloudformation/current.go]: This seems like a good idea, I am just not sure if our resources execute fine if only the name is set.

Comment (xh3b4sd) [reply service/controller/v18/resource/cloudformation/current.go]: They do, except some error handling. Which I am fixing still. Stay tuned.

**Comment by xh3b4sd** [comment]
Works on `ginger`. I figured there are other problems with the `statusresource` due to the symptoms we saw yesterday already caused by some weird `operatorkit` behaviour. I want to get after this next as well.

File: service/controller/v18/resource/cloudformation/current.go [hunk 2]
```diff
@@ -56,8 +56,22 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found stack %#q in the AWS API", stackName))
 				r.logger.LogCtx(ctx, "level", "debug", "message", "keeping finalizer")
 				finalizerskeptcontext.SetKept(ctx)
+
+				break
```
[the remark this answers is not in the document]
Comment (xh3b4sd) [reply service/controller/v18/resource/cloudformation/current.go]: Having the logs makes sense. Done.

**Pull request merged**